lib/internal/fs/rimraf.js

```diff
@@ -174,6 +194,24 @@ function rimrafPromises(path, options) {
 function rimrafSync(path, options) {
   let stats;
 
+  try {
+    stats = statSync(path);
+  } catch (err) {
+    if (err.code === 'ENOENT')
```

Comment:
what is swallowed here if not `ENOENT`?

Reply:
`rimraf` swallows all kinds of errors and throwing all errors from `stat` caused lots of failures.

Reply:
it looks like you'd get into trouble here if L198 threw, because `stats` would then be undefined, and the call to `isDirectory()` on L204 would fail...  it doesn't seem recoverable?